### PR TITLE
Refactoring ceil/round/floor code for datetime64[ns] types

### PR DESCRIPTION
**Pull request by mayankanand007**
This PR is a follow up to #9820 where @bdice and @vyasr raised the point of having a design such that we avoid writing bunch of boilerplate code, which is common in the implementations of ceil/round/floor. The aim is to reduce the total number of functions, as well as have a cleaner design.

### CODE REVIEW

File: python/cudf/cudf/_lib/datetime.pyx
Comment (brandon-b-miller) [inline python/cudf/cudf/_lib/datetime.pyx:68]: might as well update this

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timedelta.resolution_string.html

Comment (bdice) [reply python/cudf/cudf/_lib/datetime.pyx:68]: Thanks for this! I was looking at [`pandas.Timedelta.resolution`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timedelta.resolution.html) which does not have any docs.

File: python/cudf/cudf/_lib/datetime.pyx
Comment (shwina) [inline python/cudf/cudf/_lib/datetime.pyx:73]:
```suggestion
    elif freq in ("T", "min"):
```

ditto below

**Review by shwina — APPROVED** [review]
Approving with a minor suggestion.

File: python/cudf/cudf/_lib/datetime.pyx
Comment (bdice) [reply python/cudf/cudf/_lib/datetime.pyx:73]: This was previously suggested by @vyasr — this change requires the GIL and can’t be easily done in a cdef function iirc.

Comment (mayankanand007) [reply python/cudf/cudf/_lib/datetime.pyx:73]: Yep, putting a link to that: https://github.com/rapidsai/cudf/pull/9571/files#r743999332

Comment (mayankanand007) [reply python/cudf/cudf/_lib/datetime.pyx:73]: Just tried it with this suggestion and it works! I guess remove the gill made this possible @bdice!

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

**Review by bdice — APPROVED** [review]
Thanks @mayankanand007! This currently has 2 approvals. I did a final pass and I think it's ready to merge. I appreciate your hard work!

**Comment by bdice** [comment]
@gpucibot merge

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**
